src/server/scripts/Maelstrom/kezan.cpp

```diff
@@ -13,6 +13,186 @@
+    RADIO_SOUND                     = 23406
+};
+
+// 66298 - Honk Horn
+class spell_hot_rod_honk_horn : public SpellScript
+{
+    PrepareSpellScript(spell_hot_rod_honk_horn);
+
+    void Beep(SpellEffIndex /*effIndex*/)
+    {
+        Player* player = nullptr;
+
+        if (Unit* caster = GetCaster())
+            if (Vehicle* vehicle = caster->GetVehicleKit())
+                if (Unit* passenger = vehicle->GetPassenger(SEAT_PLAYER))
+                    player = passenger->ToPlayer();
+
+        if (!player)
+            return;
+    }
+
+    void Register() override
+    {
+        OnEffectHit += SpellEffectFn(spell_hot_rod_honk_horn::Beep, EFFECT_0, SPELL_EFFECT_PLAY_SOUND);
+    }
+};
+
+// 66299 - Radio
+class spell_hot_rod_radio : public SpellScript
+{
+    PrepareSpellScript(spell_hot_rod_radio);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        radioOn = false;
+        return true;
+    }
+
+    void HandleOnHit()
+    {
+        if (radioOn)
+            return;
+
+        if (Unit* caster = GetCaster()->GetOwner())
+        {
+            if (caster->GetTypeId() == TYPEID_PLAYER)
+                GetCaster()->PlayDistanceSound(RADIO_SOUND, caster->ToPlayer());
+
+            radioOn = true;
+        }
+    }
+private:
+    bool radioOn;
+
+    void Register() override
+    {
+        OnHit += SpellHitFn(spell_hot_rod_radio::HandleOnHit);
+    }
+};
+
+// 66617 - (Serverside/Non-DB2) Rolling with my Homies: Get Into The Hot Rod Trigger
+class spell_hot_rod_periodic_check : public AuraScript
+{
+    PrepareAuraScript(spell_hot_rod_periodic_check);
+
+    void HandleDummyTick(AuraEffect const* /*aurEff*/)
+    {
+        Unit* caster = GetCaster();
+
+        if (caster->GetVehicle())
+            return;
+
+        caster->CastSpell(caster, SPELL_SERVERSIDE_HOT_ROD_MOUNT, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_hot_rod_periodic_check::HandleDummyTick, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 66618 - (Serverside/Non-DB2) Rolling with my Homies: Get Into The Hot Rod Effect
+class spell_hot_rod_mount : public SpellScript
+{
+    PrepareSpellScript(spell_hot_rod_mount);
+
+    bool Load() override
+    {
+        return GetCaster()->GetOwner();
+    }
+
+    void HandleMountHotRod(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Creature* vehicle = GetCaster()->GetOwner()->ToPlayer()->GetVehicleCreatureBase();
+        CastSpellExtraArgs args;
+
+        switch (caster->GetEntry())
+        {
+            case NPC_SUMMONED_ACE:
+                args.AddSpellBP0(SEAT_ACE);
+                break;
+            case NPC_SUMMONED_GOBBER:
+                args.AddSpellBP0(SEAT_GROBBER);
+                break;
+            case NPC_SUMMONED_IZZY:
+                args.AddSpellBP0(SEAT_IZZY);
+                break;
+        }
+
+        caster->CastSpell(vehicle, VEHICLE_SPELL_RIDE_HARDCODED, args);
```

Comment:
I'm not exactly sure if we should be forcing this spell to be cast, but rather access the vehicle so we can mount it up.

Reply:
this spell is in the sniff

Reply:
Not saying otherwise, but we have helpers and functions in place that already force the spell to be cast when, e.g., clicking on a mountable npc.